src/algorithms/reference-value.ts

```diff
@@ -0,0 +1,55 @@
+import { ReferenceResult } from '../types';
+const _ = require('lodash');
+
+export function ReferenceValue(
+  x: number,
+  formulae: Array<{
```

Comment:
We can create a type named `Formula` in this file and here we can set the type of this argument to `Formula[]`